### ISSUE TITLE
purchases_flutter >=3.9.1 depends on freezed_annotation ^1.1.0

### ISSUE THREAD

**Issue by geoffn** [RevenueCat/purchases-flutter#375]
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.4 21F79 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.67.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

**Describe the bug**
Maybe less of a bug and more of a request to upgrade to latest version of freezed_annotation.  I require this for other tools and can not upgrade to purchases_flutter 3.9.0 or later. 

Because purchases_flutter >=3.9.1 depends on freezed_annotation ^1.1.0 and MyProject depends on freezed_annotation ^2.0.1, purchases_flutter >=3.9.1 is forbidden.

**Comment by julienlebren** [comment]
That's definitely a serious blocking point, I had to downgrade a lot of packages to make `purchases_flutter` work due to this.

**Comment by vegaro** [comment]
We merged https://github.com/RevenueCat/purchases-flutter/pull/363#discussion_r867034637 this week and it will be released with purchases-flutter 4.0.0. Before that, we need to do a bunch of testing, since it's a major release, and last time we made a change to freezed, we shipped a couple bugs that we didn't really expect at all.

In any case, 4.0.0 is almost ready and should be released within the next couple weeks.
